resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/config.scala

```diff
@@ -269,12 +269,41 @@ package object config extends Logging {
       .createWithDefaultString("1s")
 
   // Spark resource staging server.
+  private[spark] val RESOURCE_STAGING_SERVER_API_SERVER_URL =
+    ConfigBuilder("spark.kubernetes.resourceStagingServer.apiServer.url")
+      .doc("URL for the Kubernetes API server. The resource staging server monitors the API" +
+        " server to check when pods no longer are using mounted resources. Note that this isn't" +
+        " to be used in Spark applications, as the API server URL should be set via spark.master.")
```

Comment:
It would be difficult to specify a concise API for a request to "watch this resource", I think. Do we provide an API specifying a resource name, resource type, and labels?

The problem with having only reset-on-access is it leaves out the possibility that entities like pods might only re-run after a long period of time, certainly longer than the expiration window for the resource. This is why I think the watch-based behavior is more or less mandatory to get the most consistent behavior.